convert2rhel/pkgmanager/handlers/yum.py

```diff
@@ -134,19 +134,22 @@ def _perform_operations(self):
 
         loggerinst.info("Adding %s packages to the yum transaction set.", system_info.name)
 
-        for pkg in original_os_pkgs:
-            self._base.update(pattern=pkg)
-            try:
-                self._base.reinstall(pattern=pkg)
-            except (pkgmanager.Errors.ReinstallInstallError, pkgmanager.Errors.ReinstallRemoveError):
+        try:
+            for pkg in original_os_pkgs:
+                self._base.update(pattern=pkg)
                 try:
-                    self._base.downgrade(pattern=pkg)
-                except (
-                    pkgmanager.Errors.ReinstallInstallError,
-                    pkgmanager.Errors.ReinstallRemoveError,
-                    pkgmanager.Errors.DowngradeError,
-                ):
-                    loggerinst.warning("Package %s not available in RHEL repositories.", pkg)
+                    self._base.reinstall(pattern=pkg)
+                except (pkgmanager.Errors.ReinstallInstallError, pkgmanager.Errors.ReinstallRemoveError):
+                    try:
+                        self._base.downgrade(pattern=pkg)
+                    except (
+                        pkgmanager.Errors.ReinstallInstallError,
+                        pkgmanager.Errors.ReinstallRemoveError,
+                        pkgmanager.Errors.DowngradeError,
+                    ):
+                        loggerinst.warning("Package %s not available in RHEL repositories.", pkg)
+        except pkgmanager.Errors.NoMoreMirrorsRepoError:
+            loggerinst.critical("Failed to determine mirrors for transaction.")
```

Comment:
If we fail to determine the mirrors for the package setup, we might raise a critical and start the rollback.